libcxx/include/__ranges/concat_view.h

```diff
@@ -0,0 +1,646 @@
+            return __base_iter(in_place_index<_Indices>, std::get<_Indices>(std::move(__src_var)));
+          }...};
+          return __vtable[__idx](__src);
+        }(__i.index(), make_index_sequence<variant_size_v<__base_iter>>{})),
+        __parent_(__i.parent_) {}
+
+  _LIBCPP_HIDE_FROM_ABI constexpr decltype(auto) operator*() const {
+    return std::visit(
+        [](auto&& __it) -> __concat_reference_t<__maybe_const<_Const, _Views>...> { return *__it; }, __it_);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr __iterator& operator++() {
+    size_t __active_index = __it_.index();
+    __apply_at_index<std::variant_size_v<decltype(__it_)>>(__active_index, [&](auto __index_constant) {
+      constexpr size_t __i = __index_constant.value;
+      ++std::get<__i>(__it_);
+      __satisfy<__i>();
+    });
+    return *this;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr void operator++(int) { ++*this; }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr __iterator operator++(int)
+    requires(forward_range<__maybe_const<_Const, _Views>> && ...)
+  {
+    auto __tmp = *this;
+    ++*this;
+    return __tmp;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr __iterator& operator--()
+    requires __concat_is_bidirectional<_Const, _Views...>
+  {
+    size_t __active_index = __it_.index();
+    __apply_at_index<std::variant_size_v<decltype(__it_)>>(__active_index, [&](auto __index_constant) {
+      constexpr size_t __i = __index_constant.value;
+      __prev<__i>();
+    });
+    return *this;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr __iterator operator--(int)
+    requires __concat_is_bidirectional<_Const, _Views...>
+  {
+    auto __tmp = *this;
+    --*this;
+    return __tmp;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr bool operator==(const __iterator& __x, const __iterator& __y)
+    requires(equality_comparable<iterator_t<__maybe_const<_Const, _Views>>> && ...)
+  {
+    return __x.__it_ == __y.__it_;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr decltype(auto) operator[](difference_type __n) const
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    return *((*this) + __n);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr __iterator operator+(const __iterator& __it, difference_type __n)
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    auto __temp = __it;
+    __temp += __n;
+    return __temp;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr __iterator operator+(difference_type __n, const __iterator& __it)
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    return __it + __n;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr __iterator& operator+=(difference_type __n)
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    size_t __active_index = __it_.index();
+    if (__n > 0) {
+      std::visit(
+          [&](auto& __active_it) {
+            __apply_at_index<std::tuple_size_v<decltype(__parent_->__views_)>>(
```

Comment:
```suggestion
            __apply_at_index<tuple_size_v<decltype(__parent_->__views_)>>(
```

Ditto for other `std::tuple_size_v`.